storagenode/contact/chore.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package contact
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/internal/sync2"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/storagenode/trust"
+)
+
+var mon = monkit.Package()
+
+// Config contains configurable parameters for contact chore
+type Config struct {
+	Interval time.Duration `help:"how frequently the node contact chore should run" releaseDefault:"1h" devDefault:"30s"`
+	MaxSleep time.Duration `help:"maximum duration to wait before pinging satellites" releaseDefault:"45m" devDefault:"0s" hidden:"true"`
+}
+
+// Chore is the contact chore for nodes announcing themselves to their trusted satellites
+type Chore struct {
+	log       *zap.Logger
+	self      overlay.NodeDossier
+	transport transport.Client
+
+	trust *trust.Pool
+
+	maxSleep time.Duration
+	Loop     *sync2.Cycle
```

Comment:
`sync2.Cycle` are exposed for testing, such that tests can pause and trigger them on will, rather than using `Sleeping`. Ideally, yes, they would be hidden, but for testing, it's easier to expose them rather than build a different mechanism for exposing them only for tests.